src/client/dfs/dfs.c

```diff
@@ -1514,14 +1514,14 @@ dfs_mkdir(dfs_t *dfs, dfs_obj_t *parent, const char *name, mode_t mode,
 	if (rc)
 		return rc;
 
+	D_ASSERT(parent != NULL);
```

Comment:
in the comments, "  \param[in]	parent	Opened parent directory object. If NULL, use root obj."
maybe confirm with @mchaarawi

Reply:
the assert is OK here because we should have replaced the parent with the root one at this point (check on line 1509)